timesketch/lib/analyzers/hashr_lookup.py

```diff
@@ -0,0 +1,317 @@
+                           ' Set chunksize to default value of 50,000.')
+            self.query_chunk_size = 50000
+        except TypeError:
+            logger.warning('HASHR_QUERY_CHUNK_SIZE is not set.'
+                           ' Set chunksize to default value of 50,000.')
+            self.query_chunk_size = 50000
+
+        if (db_user is None or db_pass is None or db_address is None
+                or db_port is None or db_name is None
+                or self.add_source_attribute is None):
+            msg = ('The hashR analyzer is not able to load the hashR database '
+                   'information from the timesketch.conf file. Please make sure'
+                   ' to uncomment the section and provide the required '
+                   'connection details!')
+            sys.tracebacklimit = 0
+            raise Exception(msg)
+
+        db_string = (f'postgresql://{db_user}:{db_pass}@'
+                     f'{db_address}:{db_port}/{db_name}')
+        db_string_redacted = (f'postgresql://{db_user}:***@'
+                              f'{db_address}:{db_port}/{db_name}')
+        db = sqla.create_engine(
+            db_string, connect_args={'connect_timeout': 10})
+        try:
+            db.connect()
+            logger.info('Successful connected to hashR postgress database: %s',
+                        db_string_redacted)
+            self.hashr_conn = db
+            return True
+        except sqla.exc.OperationalError as err:
+            logger.error(
+                '!!! Connection to the hashR postgres database not '
+                'possible! -- Provided connection string: "%s"'
+                ' -- Error message: %s', db_string_redacted, str(err))
+            return ('Connection to the database FAILED. Please check the celery'
+                    ' logs and make sure you have provided the correct database'
+                    ' information in the analyzer file!')
+
+    def check_against_hashR(self, sample_hashes):
+        """Check a list of hashes against the hashR database.
+
+        Args:
+          sample_hashes:  A list of hash values that shall be checked
+                          against the hashR database.
+
+        Returns:
+          matching_hashes:  A dict containing all hashes that are found in the
+                            hashR database. (If add_source_attribute is True
+                            it also contains the corresponding source images.)
+          None:             If there are no matches, the function returns None.
+
+        Raises:
+          Exception:  Raises an exception if the provided sample_hashes
+                      parameter is not of type list or dict.
+        """
+        if not isinstance(sample_hashes, list):
+            raise Exception('The check_against_hashR function only accepts '
+                            'type<list> as input. But type '
+                            f'{type(sample_hashes)} was provided!')
+
+        chunked_hashes_list = list()
```

Comment:
I like the approach using generators to prevent building multiple huge lists in memory. However, I ran into one problem with generators in this specific use-case: When I build and send the sql query I need a full list of the full batch of hashes I want to query. An iterator on a generator does not work here. So I have to build at least one list of hashes with batch-size anyway.  
So I consider the following solution as the optimal approach which limits it to one list with maximal batch-size per iteration:
```
def batch_hashes(hash_list, batch_size=DEFAULT_BATCH_SIZE):
    for i in range(0, len(hash_list), batch_size):
        yield hash_list[i : i + batch_size]
            
for batch in self.batch_hashes(sample_hashes, self.query_batch_size):
    # query batch against postgres db
```